x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java

```diff
@@ -780,6 +786,23 @@ public void testKibanaSystemRole() {
             assertThat(kibanaRole.indices().allowedIndicesMatcher(UpdateSettingsAction.NAME).test(indexAbstraction), is(false));
             assertThat(kibanaRole.indices().allowedIndicesMatcher(RolloverAction.NAME).test(indexAbstraction), is(false));
         });
+
+        // Ensure privileges necessary for ILM policies in APM & Endpoint packages
+        Arrays.asList("metrics-apm.app-*", "metrics-apm.internal-*", "metrics-apm.profiling-*", "logs-apm.error_logs-*", "traces-apm-*")
+            .forEach(indexName -> {
+                logger.info("index name [{}]", indexName);
+                final IndexAbstraction indexAbstraction = mockIndexAbstraction(indexName);
+
+                assertThat(kibanaRole.indices().allowedIndicesMatcher(UpdateSettingsAction.NAME).test(indexAbstraction), is(true));
+                assertThat(kibanaRole.indices().allowedIndicesMatcher(RolloverAction.NAME).test(indexAbstraction), is(true));
```

Comment:
These are already granted by the overall wider privileges for `metrics-*` and `traces-*`, but I added an explicit test to ensure we meet the ILM requirements